akka-projection-jdbc/src/main/scala/akka/projection/jdbc/javadsl/JdbcProjection.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc.javadsl
+
+import java.sql.Connection
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+import akka.Done
+import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import akka.projection.Projection
+import akka.projection.ProjectionId
+import akka.projection.javadsl.SourceProvider
+import akka.projection.javadsl.HandlerLifecycle
+import akka.japi.function.{ Function => JFunction }
+
+object JdbcProjection {
+
+  def exactlyOnce[Offset, Envelope, S <: JdbcSession](
+      projectionId: ProjectionId,
+      sourceProvider: SourceProvider[Offset, Envelope],
+      sessionFactory: () => S,
+      handler: JdbcHandler[Envelope, S]): JdbcProjection[Envelope] = ???
+
+}
+trait JdbcProjection[Envelope] extends Projection[Envelope] {
+
+  /**
+   * For testing purposes the offset table can be created programmatically.
+   * For production it's recommended to create the table with DDL statements
+   * before the system is started.
+   */
+  def createOffsetTableIfNotExists(system: ActorSystem[_]): CompletionStage[Done]
+}
+
+object JdbcSession {
+
+  /**
+   * INTERNAL API: run the blocking DB operations on a single thread one the passed EC.
+   */
+  @InternalApi
+  private[akka] def runBlockingDBIO[S <: JdbcSession, Result](jdbcSessionFactory: () => S)(block: S => Result)(
```

Comment:
This is the place were we should run all the JDBC operation, including creating a JdbcSession, commit and close the connection.